.gitlab-ci.yml

```diff
@@ -218,6 +207,101 @@ build_windows_msi_x64:
     paths:
     - .omnibus/pkg
 
+
+
+#
+# image_build
+#
+
+.dind_job_template: &dind_job_definition
+  stage: image_build
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/build:latest
+  tags: [ "runner:dind", "size:large" ]
+  script: [ "# noop" ]
+
+# build the agent6 image
+build_agent6:
+  <<: *dind_job_definition
+  variables:
+    DD_DIND_BUILD_CONTEXT: Dockerfiles/agent6
+    DD_DIND_IMAGE: &agent_ecr 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/agent:$CI_PIPELINE_ID
+    DD_DIND_ARTEFACTS: "true"
+
+# build the dogstatsd image
+build_dogstatsd:
+  <<: *dind_job_definition
+  variables:
+    DD_DIND_BUILD_CONTEXT: Dockerfiles/dogstatsd/alpine
+    DD_DIND_IMAGE: &dogstatsd_ecr 486234852809.dkr.ecr.us-east-1.amazonaws.com/datadog-agent/dogstatsd:$CI_PIPELINE_ID
+    DD_DIND_ARTEFACTS: "true"
+
+#
+# image_deploy
+#
+
+.dind_tag_job_template: &dind_tag_job_definition
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/devtools/dind/tag:latest
+  tags: [ "runner:dind", "size:large" ]
+  stage: image_deploy
+  script: [ "# noop" ]
+
+agent6_docker_hub:
+  <<: *dind_tag_job_definition
+  when: manual
+  only:
+    - master
+    - tags
+  variables:
+    DD_DIND_TAG_SOURCE: *agent_ecr
+    DD_DIND_DEST_REGISTRY_TYPE: DOCKER_HUB
```

Comment:
Can be factored in. I guess if the anchor already has a `variables` object the new one is merged in instead of replacing the whole section. Am I wrong?